### PR TITLE
Absolute fill

### PR DESCRIPTION
**Pull request by jo-tomlinson**
## Description
Adds the class `.u-absolute-fill`

Also adds the mixin `edge-positioning($edge-position-offset: 0)`

## Related Issue
Fixes https://github.com/sky-uk/sky-pages/issues/3446

## Motivation and Context
Motivated by the fact that we wanted to set the height and background for the responsive-image container for media-tiles before the images have fully loaded. See https://github.com/sky-uk/sky-pages/pull/3815 for more context.

## How Has This Been Tested?
Tests added for the edge-positioning mixin.

This has also been tested manually running the above sky-pages PR locally linked to this PR.

## Screenshots (if appropriate)
The height and background for the wrapper div is set before the image loads.

responsive_image

## Types of Changes
What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Browser Support
- [x] Chrome
- [x] Edge
- [x] Firefox
- [x] IE9
- [x] IE10
- [x] IE11
- [x] Opera
- [x] Safari
- [ ] Mobile Devices
- [ ] Screen-readers (e.g. JAWS, Apple VoiceOver)

## Checklist
- [x] All code includes full inline documentation (as per the [template](https://github.com/sky-uk/toolkit-core/blob/master/_template.scss)).
- [x] All code conforms to the Toolkit [coding style](https://github.com/sky-uk/toolkit/wiki/Coding-Style).
- [x] All code conforms to [WCAG 2.0 level AA Accessibility Guidelines](https://www.w3.org/TR/WCAG20/).
- [x] New functions and mixins have appropriate tests.
- [x] All new and existing tests passed.
- [x] I have added instructions on how to test my changes.
- [ ] Package version updated.
- [ ] CHANGELOG.md updated.

### CODE REVIEW

**Comment by joebell93** [comment]
Just a thought, I've been working on an edge positioning mixin for [Tiles in Toolkit-UI 2.0](https://github.com/sky-uk/toolkit-ui/blob/2a59303e73a16533fd4037386457e499302ce862/components/_tile.scss#L26)

Why don't we extract this out and make a generic mixin - then use in this class declaration? 😊

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,13 @@
 `sky-toolkit-core` follows [Semantic Versioning](http://semver.org) to help manage the impact of releasing new library versions.
 
 
+## 1.10.0
+
+### Features
+
+* [mixins] edge-positioning mixin to simplify and avoid repeated use of individual edge positioning.
+* [utilities] addition of `u-absolute-fill` to absolutely position an element to fully fill the parent container.
```
Comment (joebell93) [inline CHANGELOG.md]: Just a thought, I think this should actually be `.u-fill-absolute` - what do you reckon?

File: test/tools/_mixins.scss [hunk 2]
```diff
@@ -153,3 +153,48 @@
...
+
+// Edge Positioning mixin
+// ===========================================
+
+@include test-module("@mixin edge-positioning") {
+  @include test("when no arguments are given to the edge-positioning mixin") {
+    @include assert("should offest all positions by zero ") {
+      @include input() {
+        .foo {
+          @include edge-positioning();
```
Comment (joebell93) [inline test/tools/_mixins.scss]: Could we call it `edge-position()`? :1st_place_medal:

File: utilities/_fill.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+   ========================================================================== */
+
+/**
+ * Utility classes to position an element within its parent.
+ */
+
+.u-absolute-fill {
+  position: absolute;
+
+  @include edge-positioning();
```
Comment (joebell93) [inline utilities/_fill.scss]: How can the mixin handle multiple values for shorthand?

e.g. `edge-position(4px, 5px)`?

File: CHANGELOG.md (hunk 1 above)
Comment (jo-tomlinson) [inline CHANGELOG.md]: i will change

File: test/tools/_mixins.scss (hunk 2 above)
Comment (jo-tomlinson) [inline test/tools/_mixins.scss]: Sure!

File: utilities/_fill.scss (hunk 3 above)
Comment (jo-tomlinson) [inline utilities/_fill.scss]: It doesn't - the other version doesn't & I thought we were just incorporating that. It could be changed if required.

Comment (joebell93) [inline utilities/_fill.scss]: Nah this is great for now! Let's look at adding this functionality in the future

File: test/tools/_mixins.scss [hunk 4]
```diff
@@ -153,3 +153,48 @@
     }
   }
 }
+
+// Edge Position mixin
+// ===========================================
+
+@include test-module("@mixin edge-positioning") {
```
Comment (joebell93) [inline test/tools/_mixins.scss]: Don't forget to change references of `edge-positioning` here to `edge-position` too

**Comment by creditlte** [comment]
@jo-tomlinson @joebell93 @steveduffin Hello all, could one you kindly help removing me from these repositories as I no longer work with Sky but I'm still getting notifications from your work. Just don't want to end up in court :wink:

**Comment by joebell93** [comment]
@creditlte that's because Toolkit's public :) just click "unwatch" at the top of the repo.
